picasso/src/main/java/com/squareup/picasso/ImageViewAction.java

```diff
@@ -44,6 +44,8 @@
       return;
     }
 
+    result.prepareToDraw();
```

Comment:
At least on line 176 we still don't know if we have a result. This should be done if we know a result was received e.g. on line https://github.com/square/picasso/blob/a834e259fba01971651a3cd684ea5370f3769de0/picasso/src/main/java/com/squareup/picasso/BitmapHunter.java#L180 instead.

Furthermore, we could minimize the window to only the point where we know the bitmap will have a high chance of actually being rendered. 

Perhaps right before we batch the results to send them to main thread here https://github.com/square/picasso/blob/master/picasso/src/main/java/com/squareup/picasso/Dispatcher.java#L430